astropy/utils/data.py

```diff
@@ -1125,46 +1167,61 @@ def is_url_in_cache(url_key):
     Returns
     -------
     in_cache : bool
-        `True` if a download from ``url_key`` is in the cache
+        `True` if a download from ``url_key`` is in the cache, `False` if not
+        or if the cache does not exist at all.
+
+    See Also
+    --------
+    cache_contents : obtain a dictionary listing everything in the cache
     """
-    # The code below is modified from astropy.utils.data.download_file()
-    try:
-        dldir, urlmapfn = _get_download_cache_locs()
-    except OSError as e:
-        msg = 'Remote data cache could not be accessed due to '
-        estr = '' if len(e.args) < 1 else (': ' + str(e))
-        warn(CacheMissingWarning(msg + e.__class__.__name__ + estr))
-        return False
+    with _cache() as (dldir, url2hash):
+        return url_key in url2hash
 
-    with shelve.open(urlmapfn) as url2hash:
-        if url_key in url2hash:
-            return True
-    return False
 
+def cache_total_size():
+    """Return the total size in bytes of all files in the cache."""
+    with _cache() as (dldir, url2hash):
+        return sum(os.path.getsize(os.path.join(dldir, h))
```

Comment:
```suggestion
        from astropy import units as u  # Put import here to avoid potential circular import
        return sum(os.path.getsize(os.path.join(dldir, h))
```